src/interfaces/token.ts

```diff
@@ -191,157 +187,76 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
   }
 
   /**
-   * Fetches supported zapper tokens and sets their icon
-   * @returns zapper tokens with icons
+   * Fetch the token amount that spender is allowed to spend on behalf of owner
+   * @param ownerAddress
+   * @param tokenAddress
+   * @param spenderAddress
+   * @returns TokenAllowance
    */
-  private async getZapperTokensWithIcons(): Promise<Token[]> {
-    const zapperTokens = await this.yearn.services.zapper.supportedTokens();
-
-    const zapperTokensAddresses = zapperTokens.map(({ address }) => address);
-
-    const zapperTokensIcons = await this.yearn.services.asset.ready.then(() =>
-      this.yearn.services.asset.icon(zapperTokensAddresses)
-    );
-
-    const setIcon = (token: Token): Token => {
-      const icon = zapperTokensIcons[token.address];
-      return icon ? { ...token, icon } : token;
-    };
-
-    return zapperTokens.map(setIcon);
-  }
-
-  /**
-   * Get allowance for a token either zapIn or direct deposit
-   * @param vault
-   * @param token
-   * @param amount
-   * @param account
-   * @returns transaction
-   */
-  async allowance(address: Address, vaultToken: Address, token: Address, account: Address): Promise<TokenAllowance> {
-    // If Ether is being sent, no need for approval
+  async allowance(ownerAddress: Address, tokenAddress: Address, spenderAddress: Address): Promise<TokenAllowance> {
     const allowance: TokenAllowance = {
-      owner: account,
-      spender: address,
-      amount: MaxUint256.toString(),
-      token
+      owner: ownerAddress,
+      token: tokenAddress,
+      spender: spenderAddress,
+      amount: MaxUint256.toString()
     };
 
-    if (EthAddress === token) return allowance;
-
-    if (vaultToken === token) {
-      const tokenContract = new Contract(token, TokenAbi, this.ctx.provider.read);
-      const partnerAddress = await this.yearn.services.partner?.address;
-      const addressToApprove = (this.shouldUsePartnerService(address) && partnerAddress) || address;
-      const allowanceAmount = await tokenContract.allowance(account, addressToApprove);
-
-      return {
-        ...allowance,
-        spender: addressToApprove,
-        amount: allowanceAmount.toString()
-      };
-    }
+    if (isNativeToken(tokenAddress) || tokenAddress === spenderAddress) return allowance;
```

Comment:
in which case would tokenAddress === spenderAddress?

Reply:
You can see that `vault.getWithdrawSpenderAddress` returns the vault token address when trying to withdraw to underlying, since the vault contract will be executing the tx. When you are trying to withdraw from a vault to its underlying token, you dont really need to approve, same happens when trying to approve a native token, so I applied same hack just so that it does not throw an error.